### ISSUE TITLE
Append into file

### ISSUE THREAD

**Issue by lmerega** [eltos/PasteIntoFile#41]
Appendo into same file (text only).

Sometimes I need to add some information to a text file.
It would be nice a contextual menu to append clipboard to an existing text file.
Do you think it could be useful?

Thanks

Luca

**Comment by lmerega** [comment]
Hallo, yes, I meant any text file. It should be easy opening the file for append and adding the text.
Regarding the context menu, you're right. There are already two entries. It's true that adding another one isn't too serious, especially since I'm not a fan of tree-style context menus. However, we could also consider that solution, perhaps including it in the options: tree-style context menu or "flat" context menu.

**Comment by eltos** [comment]
How about making the entries into a sub-menu only if the user decides to enable 3 or more?

Another idea could be to re-use the "replace" entry and use, SHIFT key to switch from replace to append mode, since that is very similar in functionality.
It's not that intuitive, but SHIFT and CTRL are already used to change the behaviour of the "paste into file" menu entry.

**Comment by hirenshah** [comment]
I came across this program and then this issue when looking for the same thing.  

When researching things online rather than opening each link in a separate tab or going off in tangents, it would be perfect if I could just start a program running which until I turn off, appends all copied text to a .txt file.  It could potentially be enhanced to provide the option of prefixing each entry with a date time stamp.

**Comment by eltos** [comment]
> Start a program running which until I turn off, appends all copied text to a .txt file.

Sounds like you need the append option for what is called "Batch mode", not another context menu entry. As of now, batch mode will create a new file with timestamp every time you press CTRL+C.

**Comment by eltos** [comment]
I am thinking of the following:
- Add checkbox "Append" (non-persistent setting, default off, put it where the "Clear clipboard" checkbox is and drop that unnecessary functionality)
  - This will only work for text-like formats, other formats will show an error messages with the option to replace or cancel
  - In combination with batch mode
    - The filename field stays editable, and is _not_ updated from the template when something new is copied
    - Instead, the text is appended into the file on every copy 
    - If something is copied which does not match the selected extension, it is ignored (preview is updated but nothing saved)
    - if the user selects an extension that does not support appending, show the above error messages 
- Context menu entry "Replace with clipboard content" becomes "Paste into file" with two sub-items: Replace and Append
  - Nice-to-have: For some file formats known to not support appending stuff (e.g. PNG, JPG, ...) only show a top-level "Paste into file (replace)" entry
  - Replace is the existing functionality (no file-exists-warning, old file is moved to bin and can be restored with undo as usual)
  - Append does what it says
  - If append is selected, but it's not text-based, show error messages with the option to replace or cancel (same as above)
  - Nice-to-have: SHIFT can be used to show the dialog (default is always to skip, independent of autosave setting)

Any comments? @lmerega @hirenshah

**Comment by eltos** [comment]
Preview version at https://github.com/eltos/PasteIntoFile/actions/runs/7746499182/artifacts/1213166259

**Comment by hirenshah** [comment]
@eltos

I had disabled the context menu options from the start because my use case had no use for them. Can't really add any value around that functionality, at least not until I have had a play around. Currently in the middle of a house move so give me a few days to do that. 


Regarding batch mode:

- I tried to make your application work for my use case by removing variables from the template string. That resulted in the Windows replace file prompt coming up every time Ctrl+C was pressed. Should there be a replace file checkbox as well, or should there be validation that requires variables to be present?

- I ended up putting the variables back in and then just merging the many txt files using a batch script. 

- I like the idea of an append checkbox. Whilst it wouldn't update from the template string, what would happen if someone added a variable to the filename? 

- How is "does not match the selected extension" determined? I'm assuming it is by MIME type rather than actual extension?

**Comment by eltos** [comment]
@hirenshah I see, it seams this is outdated information now. Maybe you first want to give that preview version a try and then update your feedback?

**Comment by hirenshah** [comment]
> @hirenshah I see, it seams this is outdated information now. Maybe you first want to give that preview version a try and then update your feedback?
> 

Yep, I was just annoyed about not having posted it and also didn't want you to think you were being ignored 🙂

Will check out the new version and report back... will make sure to actually report back lol.

**Comment by eltos** [comment]
New preview at https://github.com/eltos/PasteIntoFile/actions/runs/8316160819/artifacts/1333099887

Scheduled for 5.3 release
